### PR TITLE
Add: Project scopes

### PR DESCRIPTION
**Pull request by darshan487**
### **PR Type**
enhancement

___

### **Description**
- Added multiple TypeScript functions to manage project scopes, including creating, updating, deleting, and fetching scopes.
- Implemented member assignment and unassignment to scopes.
- Introduced batch deletion and moving scopes to new activity phases.
- Provided example usages for each function to demonstrate functionality.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>assign_member_to_scope.ts</strong><dd><code>Add member assignment to project scope functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/assign_member_to_scope.ts

<li>Added function to assign or unassign a member to a project scope.<br> <li> Defined <code>AssignMemberParams</code> interface for function parameters.<br> <li> Included example usage for assigning and unassigning a member.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-deebbbf8350b57454611c49dbe1aeff0a230f04a01e36203f70a101154672242">+39/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>create_one.ts</strong><dd><code>Implement project scope creation functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/create_one.ts

<li>Added function to create a new project scope.<br> <li> Defined <code>CreateScopeParams</code> interface for function parameters.<br> <li> Included example usage for creating a project scope.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-c9a2d17ce03c6bf2608f7dfc04eb82da59d27b4d9fdaa25fba8a2d538d828f0d">+38/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>delete_one.ts</strong><dd><code>Add single project scope deletion functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/delete_one.ts

<li>Added function to delete a single project scope.<br> <li> Included example usage for deleting a project scope.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-58a337f36a6c998ba337eaf817ffc65bdfb5c990e45f50ae2e72fd93e9176446">+19/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>delete_scopes_in_batch.ts</strong><dd><code>Implement batch deletion of project scopes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/delete_scopes_in_batch.ts

<li>Added function to delete multiple project scopes in batch.<br> <li> Defined <code>DeleteBatchScopesParams</code> interface for function parameters.<br> <li> Included example usage for batch deletion.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-52dc05894ad03462976dbbb439fed80f5728178762a0a17738fff733830161f9">+24/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>fetch_all.ts</strong><dd><code>Add functionality to fetch all project scopes</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/fetch_all.ts

<li>Added function to fetch all project scopes with optional filters.<br> <li> Defined <code>FetchScopesParams</code> interface for query parameters.<br> <li> Included example usage for fetching scopes.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-8654c9141295aa1f88ea2be40c579ae2c3bcf7aaccdd60fda523d7e11d026621">+46/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>move_scope_to_new_activity_phase.ts</strong><dd><code>Implement moving scopes to new activity phase</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/move_scope_to_new_activity_phase.ts

<li>Added function to move scopes to a new activity phase.<br> <li> Defined <code>MoveScopesParams</code> interface for function parameters.<br> <li> Included example usage for moving and copying scopes.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-becace169db8fbc5adb5ac28548082824f44b0f182ec146a0468494d5a2d5855">+34/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>update_assign_member_to_scope.ts</strong><dd><code>Add functionality to update scope assignment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/update_assign_member_to_scope.ts

<li>Added function to update scope assignment details.<br> <li> Defined <code>UpdateAssignmentParams</code> interface for query parameters.<br> <li> Included example usage for updating assignments.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-cd56fcab4c7a1c73a69e26618a231f222f7885754f25b15bdbf102635051bc45">+32/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>update_one.ts</strong><dd><code>Implement project scope update functionality</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

project_scopes/update_one.ts

<li>Added function to update project scope details.<br> <li> Defined <code>UpdateScopeParams</code> interface for function parameters.<br> <li> Included example usage for updating scope fields.<br>

</details>

  </td>
  <td><a href="https://github.com/LifeCoded/mosaic-api-examples/pull/46/files#diff-ef43de354beb6bfd150166f1dd0f001ee95b5275d44e4dcdb0a464d59361151e">+46/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: project_scopes/create_one.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  note?: string;
+  parent_scope_id?: number;
+  requested_to_id?: number;
+  schedule_end?: string; 
+  schedule_start?: string;
+}
+
+async function createProjectScope(
+  teamId: number,
+  postData: CreateScopeParams
```
[the remark this answers is not in the document]
Comment (darshan487) [reply project_scopes/create_one.ts:17]: No need for it

File: project_scopes/delete_one.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import api from '../api';
+
+async function deleteProjectScope(
+  teamId: number,
+  scopeId: number,
```
[the remark this answers is not in the document]
Comment (darshan487) [reply project_scopes/delete_one.ts:5]: No need for it

File: project_scopes/fetch_all.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  schedule_end?: string;
+  schedule_start?: string;
+  scope_ids?: number[];
+  sort_attributes?: string[];
+  work_plan_ids?: number[];
+}
+
+async function fetchAllScopes(
+  teamId: number,
+  queryParams: FetchScopesParams
```
[the remark this answers is not in the document]
Comment (darshan487) [reply project_scopes/fetch_all.ts:23]: No need for it

File: project_scopes/update_one.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  parent_scope_id?: number;
+  requested_to_id?: number;
+  schedule_start?: string
+  schedule_end?: string;
+}
+
+async function updateProjectScope(
+  teamId: number,
+  scopeId: number,
+  updateData: UpdateScopeParams
```
[the remark this answers is not in the document]
Comment (darshan487) [reply project_scopes/update_one.ts:16]: No need for it

File: project_scopes/delete_scopes_in_batch.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import api from '../api';
+
+interface DeleteBatchScopesParams {
+  scope_ids: number[];
+}
+
+async function deleteBatchScopes(team_id: number, data: DeleteBatchScopesParams): Promise<void> {
```
[the remark this answers is not in the document]
Comment (darshan487) [reply project_scopes/delete_scopes_in_batch.ts:7]: No need for it

**Review by LKhyeon — APPROVED** [review]

**Pull request merged**